### PR TITLE
[TA] Implement Pageable actions

### LINKED ISSUE #16958: [FEATURE REQ] Implement AsyncPageable for AnalyzeOperation
**Library or service name.**
What library or service is this request related to? 

This is for Azure.AI.TextAnalytics SDK, and we need to implement the Pagination function for AnalyzeOperation result class.

For reference please look at healthcare GetHealthcareEntities function.

**Is your feature request related to a problem? Please describe.**
What feature would you like to get added? What problem is it solving?

We need to add a pagination function using Asyncpageable, this will help us get all the results using a single function.

### PR DESCRIPTION
**Pull request by maririos**
Implements pagination for Analyze endpoint.
- Tests updated
- Test added for pagination
- Samples updated. Samples need more love so issue => https://github.com/Azure/azure-sdk-for-net/issues/17811 is for that

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/16958

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs [hunk 1]
```diff
@@ -59,25 +59,50 @@ public class AnalyzeHealthcareEntitiesOperation : PageableOperation<AnalyzeHealt
...
+        /// </summary>
         private TextAnalyticsOperationStatus _status;
 
+        /// <summary>
+        /// If the operation has an exception, this property saves its information.
+        /// </summary>
         private RequestFailedException _requestFailedException;
 
+        /// <summary>
+        /// represents the HTTP response from the service.
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs:88]: nit: capitalization `represents` => `Represents`

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeOperation.cs [hunk 2]
```diff
@@ -43,45 +38,48 @@ public class AnalyzeOperation : Operation<AnalyzeOperationResult>
...
+        /// The last HTTP response received from the server. <c>null</c> until the first response is received.
+        /// </summary>
         private Response _response;
 
-        /// <summary>The result of the long-running operation. <c>null</c> until result is received on status update.</summary>
-        private AnalyzeOperationResult _value;
+        /// <summary>
+        /// The result of the long-running operation. <c>null</c> until result is received on status update.
+        /// </summary>
+        private Page<AnalyzeOperationResult> _firstPage;
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeOperation.cs:66]: Should this comment be `Provides the results for the first page.` like in `AnalyzeHealthcareEntitiesOperation`?

**Review by kinelski — APPROVED** [review]
LGTM. Only nit comments.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeOperation.cs (hunk 2 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeOperation.cs:66]: yes! thank you!

**Review by suhas92 — APPROVED** [review]

**Pull request merged**